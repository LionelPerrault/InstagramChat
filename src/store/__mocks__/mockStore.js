import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
