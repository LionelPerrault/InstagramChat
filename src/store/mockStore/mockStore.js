import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore
}

const middlewares = [
  promise(), thunk, logger
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
