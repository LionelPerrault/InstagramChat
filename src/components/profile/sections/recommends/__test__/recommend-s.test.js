import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import mockStore from '../../../../../store/__mocks__/mockStore'
import Recommendations from '../recommends-s'
import MockDataElement from '../../../../../utils/__mocks__/mock-dataElement'

describe('Recommendations Component', () => {
  MockDataElement()

  it('should match snapshot', () => {
    const tree = create(
      <Provider store={mockStore}>
        <Recommendations param="takkar" />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
