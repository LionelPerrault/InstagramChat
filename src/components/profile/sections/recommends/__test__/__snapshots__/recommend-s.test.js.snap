// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommendations Component should match snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "animation": "bcCCNc 300ms ease 0s 1 normal none running",
        "backfaceVisibility": "visible",
      }
    }
  >
    <div
      className="senapati pro_senapati"
    >
      <div
        className="m_div"
      >
        <div
          className="m_header"
        >
          <span>
            1 recommendation
          </span>
        </div>
        <div
          className="m_wrapper"
        >
          <div
            className="m_on followers_m_on"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="m_top"
            >
              <img
                src="/users/20/avatar.jpg"
              />
              <div
                className="m_top_right"
              >
                <Link
                  to="/profile/gian"
                >
                  gian
                </Link>
                <span>
                  Takeshi
                   
                  Gauda
                </span>
              </div>
            </div>
            <div
              className="m_bottom"
            >
              <span
                className="recommend_by"
              >
                by
                <Link
                  to="/profile/selena"
                >
                   selena
                </Link>
              </span>
              <a
                className="sec_btn"
                href="#"
                onClick={[Function]}
              >
                Remove
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="page_end"
      onClick={[Function]}
    >
      <span>
        Looks like you've reached the end
      </span>
    </div>
  </div>
</div>
`;
